packages/libp2p/src/connection-manager/dial-queue.ts

```diff
@@ -464,10 +456,25 @@ export class DialQueue {
               // update dial status
               pendingDial.status = 'active'
 
-              const conn = await this.transportManager.dial(addr, {
-                ...options,
-                signal
-              })
+              let conn: Connection
+
+              try {
+                conn = await this.transportManager.dial(addr, {
+                  ...options,
+                  signal
+                })
+
+                // mark multiaddr dial as successful
+                await this._updateAddressStatus(conn.remotePeer, addr, true)
+              } catch (err: any) {
+                if (pendingDial.peerId != null) {
+                  // mark multiaddr dial as failure
+                  await this._updateAddressStatus(pendingDial.peerId, addr, false)
+                }
```

Comment:
This check implies that we may also attempt to dial peers where peerId == null. Can we log when that is the case, and that we're not marking peer as having a failed dial?